src/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable_handler.js

```diff
@@ -8,15 +8,15 @@ import { EmbeddableHandler } from 'ui/embeddable';
 import chrome from 'ui/chrome';
 
 export class VisualizeEmbeddableHandler extends EmbeddableHandler {
-  constructor($compile, $rootScope, visualizeLoader, timefilter, Notifier, Promise) {
+  constructor($compile, $rootScope, visualizeLoader, timefilter, Notifier, Promise, Private) {
     super();
     this.$compile = $compile;
     this.visualizeLoader = visualizeLoader;
     this.$rootScope = $rootScope;
     this.name = 'visualization';
     this.Promise = Promise;
     this.brushEvent = utilsBrushEventProvider(timefilter);
-    this.filterBarClickHandler = filterBarClickHandlerProvider(Notifier);
+    this.filterBarClickHandler = filterBarClickHandlerProvider(Notifier, Private);
```

Comment:
`... = Private(filterBarClickHandlerProvider); ` should work right ?

also `import { FilterBarClickHandlerProvider as filterBarClickHandlerProvider } ... ` the as part should be removed, providers should be named with uppercase, but because you are calling function here instead of passing it to private our linting rules complained that function should begin with lowercase.

Reply:
I think @Stacey-Gammon was intentionally trying to get rid of the use of Private here, is that right Stacey?

Reply:
Yes, ideally we would start passing around the parameters actually needed by all these things instead of wrapping them in Private which makes it very difficult to follow what modules need what.  Eventually this will help with extracting out business logic from angular specific logic, since the Private syntax is all angular specific.

Unfortunately, I do think this made a little more sense when there was no Private being passed in.  It does look a little strange now, passing it through.

It seems like all the map stuff uses Private only because of Promises.  Might be worth considering at some point if all the filter code could use native promises (ala https://github.com/elastic/kibana/issues/13855) and then we might be able to get rid of the need for the Private keyword at all.

Or another option is to just pass Promise if that is indeed the only class that is being grabbed by Private in all the filter code.

It does look like *a lot* of code that is using Private though, so I wasn't able to follow all the paths to double check that indeed it only uses Promise.

But in the meantime, I vote for how @Bargs has it, passing Private in as a parameter instead of wrapping it.